### PR TITLE
TGUI ATMs + QoL and Feedback Improvements

### PR DESCRIPTION
**Pull request by DisturbHerb**
[TGUI][QOL][FIX]

## About the PR 
This PR introduces a rewritten TGUI interface for the `/submachine/ATM`s that are accessible to all crew. Do note that this does NOT touch the `/machinery/computer/ATM`s that are on cog1 and the donuts.

Tab 1 of the interface.
Tab 2 of the interface.

This also adds:
- Sound effects on interactions (with hopefully adequate spam protection)
- A bug fix where Spacebux couldn't be directly transferred to other accounts, L
- Some nitpicky pedantic fixes to standardise capitalisation and punctuation in `boutput`s
- You don't uselessly tap tokens on the machine anymore it returns early
- Visual feedback on the interface when important actions occur (does not apply to the spacebux menu because i'm not sure about how to make them only appear locally)
- i think that's all

I'm gonna have to stress again that this does NOT touch `/machinery/computer/ATM` because I have no idea what to do about it. Pls halp.

## Why's this needed? 
tgui good can i go to bed yet

### CODE REVIEW

**Comment by ZeWaka** [comment]
the right thing to do with the computer/ATM is to remove and replace it

File: tgui/packages/tgui/interfaces/Atm/types.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * @file
+ * @copyright 2022
+ * @author DisturbHerb (https://github.com/DisturbHerb/)
+ * @license MIT
+ */
+
+export interface AtmData {
+  accountName: string;
+  accountBalance: number;
+	cardname: string;
```
Comment (ZeWaka) [inline tgui/packages/tgui/interfaces/Atm/types.ts:11]: eeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
```suggestion
  cardname: string;
```

File: tgui/packages/tgui/interfaces/Atm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+const TypedNoticeBox = (props) => {
+  const { type, ...rest } = props;
+  const typeProps = {
+    ...(type === Types.Danger ? { danger: true } : {}),
+    ...(type === Types.Info ? { info: true } : {}),
+    ...(type === Types.Success ? { success: true } : {}),
+  };
+  return <NoticeBox {...typeProps} {...rest} />;
+};
```
Comment (ZeWaka) [inline tgui/packages/tgui/interfaces/Atm/index.tsx:56]: why not just add the types to the actual notice box?

File: code/datums/banking.dm [hunk 3]
```diff
@@ -829,167 +847,161 @@
...
+					src.accessed_record["current_money"] -= amount
+					var/obj/item/spacecash/S = new /obj/item/spacecash
+					S.setup(src.loc, amount)
+					usr.put_in_hand_or_drop(S)
+					src.show_message("Withdrawal successful.", "success", "atm")
+					playsound(src.loc, sound_interact, 50, 1)
+					. = TRUE
 			if("withdraw_spacebux")
-				var/amount = round(input(usr, "You have [usr.client.persistent_bank] spacebux.\nHow much would you like to withdraw?", "How much?", 0) as num)
+				var/amount = round(text2num(tgui_input_text(usr, "You have [usr.client.persistent_bank] Spacebux.\nHow much would you like to withdraw?", "How much?")))
```
Comment (ZeWaka) [inline code/datums/banking.dm:983]: why not num input then

File: code/datums/banking.dm [hunk 4]
```diff
@@ -829,167 +847,161 @@
...
+				src.show_message("Log out successful. Have a secure day.", "success", "login")
+				playsound(src.loc, sound_interact, 50, 1)
+				src.scan = null
+				src.state = STATE_LOGGEDOFF
+				. = TRUE
 			if("transfer_spacebux")
 				if(!usr.client)
 					boutput(usr, "<span class='alert'>Banking system offline. Welp.</span>")
-				var/amount = input("How much do you wish to transfer? You have [usr.client.persistent_bank] spacebux", "Spacebux Transfer") as num|null
+				var/amount = text2num(tgui_input_text(usr, "How much do you wish to transfer? You have [usr.client.persistent_bank] spacebux.", "Spacebux Transfer"))
```
Comment (ZeWaka) [inline code/datums/banking.dm:942]: ditto

File: code/datums/banking.dm [hunk 5]
```diff
@@ -829,167 +847,161 @@
...
+					boutput(usr, "<span class='notice'>You swipe your ID card.</span>")
+					src.scan = O
+					. = TRUE
+			if("login_attempt")
+				if(!src.scan)
+					return FALSE
+				var/userPin
+				if (usr.mind?.remembered_pin)
+					userPin = usr.mind?.remembered_pin
+				var/enteredPIN = text2num(tgui_input_text(usr, "Enter your PIN.", src.name, userPin, 4)) // before anyone asks, yes i know tgui_input_number() exists
```
Comment (ZeWaka) [inline code/datums/banking.dm:917]: yeah so i'm asking

**Review by ZeWaka — CHANGES_REQUESTED** [review]
also why is the window so uselessly tall per your screenshot?

File: tgui/packages/tgui/interfaces/Atm/index.tsx (hunk 2 above)
Comment (DisturbHerb) [reply tgui/packages/tgui/interfaces/Atm/index.tsx:56]: IDE got angy if i tried, this is how it's implemented in the cloning console as well, would love to hear how you'd fix it

**Comment by DisturbHerb** [comment]
> the right thing to do with the computer/ATM is to remove and replace it

separate PR time

**Comment by DisturbHerb** [comment]
!merge_upstream

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Looks good to me, couple of minor things:
- Swiping a new card doesn't seem to log out the old one properly, just replaces the name:
image
(might be an old ATM bug but should be fixed regardless)
- The two spacebux buttons are scronched together with no spacing: 
image

**Comment by TobleroneSwordfish** [comment]
- Pressing cancel on the send spacebux menu results in 
image
and then a runtime. Should probably say `3 spacebux` instead of `3` too.
- The max button on the withdraw spacebux menu defaults to a million instead of however much you currently have (unlike the credits one)

**Review by TobleroneSwordfish — APPROVED** [review]

**Pull request merged**